packages/core/src/web/app/views/beambox/Right-Panels/ConfigPanel/WobbleBlock.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { memo, useContext } from 'react';
+
+import { QuestionCircleOutlined } from '@ant-design/icons';
+import { Switch, Tooltip } from 'antd';
+import classNames from 'classnames';
+
+import history from '@core/app/svgedit/history/history';
+import UnitInput from '@core/app/widgets/Unit-Input-v2';
+import { writeData } from '@core/helpers/layer/layer-config-helper';
+import { getSVGAsync } from '@core/helpers/svg-editor-helper';
+import useI18n from '@core/helpers/useI18n';
+import type ISVGCanvas from '@core/interfaces/ISVGCanvas';
+
+import styles from './Block.module.scss';
+import ConfigPanelContext from './ConfigPanelContext';
+
+let svgCanvas: ISVGCanvas;
+
+getSVGAsync((globalSVG) => {
+  svgCanvas = globalSVG.Canvas;
+});
+
+const WobbleBlock = (): React.JSX.Element => {
+  const lang = useI18n();
+  const t = lang.beambox.right_panel.laser_panel;
+  const { dispatch, initState, selectedLayers, state } = useContext(ConfigPanelContext);
+  const { wobbleDiameter, wobbleStep } = state;
+
+  const wobbleOn = wobbleStep.value > 0 && wobbleDiameter.value > 0;
+  const handleToggle = () => {
+    const newSign = wobbleOn ? -1 : 1;
+    const step = Math.abs(wobbleStep.value) * newSign;
+    const diameter = Math.abs(wobbleDiameter.value) * newSign;
+
+    dispatch({ payload: { wobbleDiameter: diameter, wobbleStep: step }, type: 'change' });
+
+    const batchCmd = new history.BatchCommand('Change wobble toggle');
+
+    selectedLayers.forEach((layerName) => {
+      writeData(layerName, 'wobbleStep', step, { batchCmd });
+      writeData(layerName, 'wobbleDiameter', diameter, { batchCmd });
+    });
+    batchCmd.onAfter = initState;
+    svgCanvas.addCommandToHistory(batchCmd);
```

Comment:
Can use `undoManager.addCommandToHistory` from `'@core/app/svgedit/history/undoManager'` directly to avoid use of `svgCanvas`